support Kerberos

em3s opened the pull request:
This PR is a trick for #1.

bryanj committed:
Set UGI configuration after initializing hadoop conf

emeth-kim committed:
Some modifications for Kerberos.

jongwook commented:
Thanks! The changes seem mostly harmless; let me add a few cosmetic changes and merge this.

jongwook committed:
cosmetic changes

emeth-kim committed:
added hadoop.security.* for One-Liner Deploy

jongwook committed:
add "hadoop.security" prefix to the defaultKeys

emeth-kim committed:
Pass the Hive configuration to `HiveConf` properly

To pass the Hive configuration to `HiveConf` properly,
all the configuration which starts with "hive." are passed by `System.setProperty`.
Then, they are read in `HiveConf.applySystemProperties`.

jongwook commented:
[SparkHadoopUtil](https://github.com/apache/spark/blob/v2.1.0/core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala#L49-L51) was reverting the UGI to the default, and I think we might not have to override `prepareLocalResources` entirely.

jongwook committed:
copy spark configurations to the system property so that SparkHadoopUtil does not pick up empty authentication settings

jongwook committed:
Hive example more readable
